server/src/main/java/com/cloud/vm/schedule/VMScheduleManagerImpl.java

```diff
@@ -0,0 +1,295 @@
+        }
+
+        vmSchedule.setState(VMSchedule.State.Disabled);
+        boolean updateResult = vmScheduleDao.update(vmSchedule.getId(), vmSchedule);
+
+        return updateResult;
+    }
+
+    @Override
+    @ActionEvent(eventType = EventTypes.EVENT_VMSCHEDULE_UPDATE, eventDescription = "update VM Schedule")
+    public VMSchedule updateVMSchedule(UpdateVMScheduleCmd cmd) {
+        VMScheduleVO vmSchedule = vmScheduleDao.findById(cmd.getId());
+        if (vmSchedule == null) {
+            throw new InvalidParameterValueException("unable to find the vm schedule with id " + cmd.getId());
+        }
+        String description = cmd.getDescription();
+        String period = cmd.getPeriod();
+        String action = cmd.getAction();
+        String tag = cmd.getTag();
+        String timezone = cmd.getTimezone();
+
+        if (vmSchedule.getState() == VMSchedule.State.Disabled) {
+            if (description != null)
+                vmSchedule.setDescription(description);
+            if (period != null)
+                vmSchedule.setPeriod(period);
+            if (action != null)
+                vmSchedule.setAction(action);
+            if (tag != null)
+                vmSchedule.setTag(tag);
+            if (timezone != null)
+                vmSchedule.setTimezone(timezone);
+
+            vmScheduleDao.update(vmSchedule.getId(), vmSchedule);
+        } else {
+            throw new InvalidParameterValueException("Enable the state of VM Schedule before updating it");
+        }
+        return vmSchedule;
+    }
+
+    public static final class VMScheduleBackgroundTask extends ManagedContextRunnable implements BackgroundPollTask {
+        private  VMScheduleManagerImpl serviceImpl;
+
+        public VMScheduleBackgroundTask(VMScheduleManagerImpl serviceImpl) {
+            this.serviceImpl = serviceImpl;
+        }
+
+        private void scheduleActionOnVM(String action, VMSchedule.State state, String uuid, String period) {
+
+            serviceImpl.vmManager.start(uuid, null);
+        }
+
+        private Boolean matchPeriodWithCurrentTimestamp(String period){
+            Date currentTimestamp = new Date();
+            String[] periodParts = period.split(" ");
+
+
+            return true;
+        }
+
+        private void getVmSchedulesToBeExecuted() {
+
+        }
+
+        @Override
+        protected void runInContext() {
+            try {
+                if (s_logger.isTraceEnabled()) {
+                    s_logger.trace("VM Scheduler GC task is running...");
+                }
+                // Date currentTimestamp = new Date();
+                for (final VMScheduleVO vmSchedule :serviceImpl.vmScheduleDao.listAll()) {
+                    VMInstanceVO vmInstance = serviceImpl.vmInstanceDao.findById(vmSchedule.getVmId());
+                    scheduleActionOnVM(vmSchedule.getAction(), vmSchedule.getState(), vmInstance.getUuid(), vmSchedule.getPeriod());
+                }
+
+                /*
```

Comment:
Minor - please remove the commented code

Reply:
removed